### PR TITLE
Ports squishing emote for slimes

### PR DESCRIPTION
**Pull request by imsxz**
:cl: imsxz
add: adds squishing emote for slimes
soundadd: adds slime_squish.ogg
/:cl:

Ports squishing emote for slimes from paradise.

TG doesn't have a species specific emotes thing set up yet and I'm not good enough at coding to figure out how to do that, so I just used can_run_emote for now.

### CODE REVIEW

**Comment by Xantholne** [comment]
NOOO, NO PLEASE

**Comment by taukausanake** [comment]
_Do it_
Also make sure that all you need is a slime body part in order to squish like they did in Paradise

**Comment by imsxz** [comment]
I tried looking for slime specific body parts, but they didn't already exist so I went with the easy option.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -77,6 +77,24 @@
...
+
+/datum/emote/living/carbon/human/squish/run_emote(mob/user, params)
+	var/mob/living/carbon/human/H = user
+	if(..() && isslimeperson(H))
+		playsound(user.loc, 'sound/effects/slime_squish.ogg', 50)
+
+/datum/emote/living/carbon/human/squish/can_run_emote(mob/user)
+	if(!..())
+		return FALSE
+	var/mob/living/carbon/human/H = user
```
Comment (FlattestGuitar) [inline code/modules/mob/living/carbon/human/emote.dm]: no need to typecase, `is_species()` already takes in all mobs

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -77,6 +77,24 @@
...
 
+/datum/emote/living/carbon/human/squish
+	key = "squish"
+	key_third_person = "squishes"
+	message = "squishes."
+	message_param = "squishes at %t."
+
+/datum/emote/living/carbon/human/squish/run_emote(mob/user, params)
+	var/mob/living/carbon/human/H = user
+	if(..() && isslimeperson(H))
```
Comment (FlattestGuitar) [inline code/modules/mob/living/carbon/human/emote.dm]: `..()` also runs the species check, so no need to repeat it

**Comment by SmokingKilz** [comment]
What's next, howling?! Skreeing?! Where does it end!!!!?

**Comment by SamHPurp** [comment]
❄️ 👎

**Comment by imsxz** [comment]
is this good now @FlattestGuitar ? not at home currently so I can't playtest

**Comment by SamHPurp** [comment]
[Add the *Squish emote for slimes](https://trello.com/c/hc4Kk14J/79-add-the-squish-emote-for-slimes)

**Pull request merged**